src/bartiq/compilation/_compile.py

```diff
@@ -11,715 +11,234 @@
-        else:
-            port = get_port_target(input_port)
-
-        param = str(port.size)
-
-        if is_constant_int(param):
-            child_param = port.absolute_path(exclude_root_name=True)
-            parent_param = input_port.absolute_path(exclude_root_name=True)
-        elif is_single_parameter(param):
-            child_param = join_paths(port.absolute_path(exclude_root_name=True), param)
-            parent_param = join_paths(input_port.absolute_path(exclude_root_name=True), param)
-        else:
-            raise ValueError("param should be either integer or a single symbol, got {param}.")
-
-        param_map[child_param] = parent_param
-
-    new_function = rename_variables(function, param_map)
-    return new_function
-
-
-def _undo_push_out_output_register_size_params(
-    function: SymbolicFunction[T_expr],
-    routine: RoutineWithFunction[T_expr],
-) -> SymbolicFunction[T_expr]:
-    """Renames the register size parameters for output ports to match their targets."""
-    # Deal with root edge case
-    if routine.is_root:
-        return function
-
-    # Define parameter map
-    param_map = {}
-    for source in routine.output_ports.values():
-        target = get_port_target(source)
-        assert target.parent is not None
-        new_param = (
-            _resolve_target_param(target) if target.parent.is_leaf else target.absolute_path(exclude_root_name=True)
-        )
-        param_map[new_param] = source.absolute_path(exclude_root_name=True)
-
-    # Return function with renamed parameters
-    return rename_variables(function, param_map)
-
-
-def _infer_missing_register_sizes(
-    function: SymbolicFunction[T_expr],
-    routine: RoutineWithFunction[T_expr],
-    backend: SymbolicBackend[T_expr],
-) -> SymbolicFunction[T_expr]:
-    """Goes and routes out any missing register sizes (which should be associated with constant register sizes)."""
-    known_register_sizes = _extract_known_register_sizes(function)
-    new_outputs = {**function.outputs}
-    for port in routine.ports.values():
-        # Skip if we've already got an expression for the size in the function
-        if port.name in known_register_sizes:
-            continue
-
-        # Otherwise, go find what the size should be, check it's a constant, and add it to the function's outputs
-        port_endpoint = _get_internal_port_endpoint(port)
-        if is_constant_int(port_endpoint.size):
-            assert port_endpoint.size  # To satisfy typechecker
-            new_output_symbol = f"#{port.name}"
-            new_outputs[new_output_symbol] = DependentVariable(
-                new_output_symbol,
-                backend.as_expression(port_endpoint.size),
-                backend=backend,
-            )
-    return SymbolicFunction(function.inputs, new_outputs)
-
+    # We start by populating it with freshly compiled local variables and inputs
+    # {None: {a: 2}, b: {foo: 3, #in_1: N}}
```

Comment:
To be honest,  this example here is somehow more confusing than revealing to me.

Reply:
I agree this example is confusing. I removed it in favor of other comment that I added near the definition of `ParameterTree`.